localstack/services/awslambda/event_source_listeners/sqs_event_source_listener.py

```diff
@@ -186,6 +190,20 @@ def delete_messages(result: InvocationResult, func_arn, event, error=None, **kwa
 
             records.append(record)
 
+        event_filter_criterias = get_lambda_event_filters_for_arn(lambda_arn, queue_arn)
+        if len(event_filter_criterias) > 0:
+            # convert to json for filtering
+            for record in records:
+                record["body"] = json.loads(record["body"])
```

Comment:
[According to the docs](https://docs.aws.amazon.com/lambda/latest/dg/invocation-eventfiltering.html#filtering-sqs), a non-JSON body would still be a valid message.

> For Amazon SQS, the message body can be any string.

As I understand from the table in the docs: If the incoming message is a plain string, and the filter pattern is a valid json, then the message should be dropped. 
Thus, I think a try-except block here would be sufficient. 

Currently [the exception is caught here and lost](https://github.com/merowinger92/localstack/blob/mk-ddbstreams-filter/localstack/services/awslambda/event_source_listeners/sqs_event_source_listener.py#L77), because we only log a message for the if-condition. 

Maybe you could log the exception in any case there 🙏